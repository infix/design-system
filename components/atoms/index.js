import Button from './Button'
import Input from './Input'
import Heading from './Heading'
import Text from './Text'
import IconButton from './IconButton'
import { RadioGroup, RadioButton } from './Radio'
import { Select } from './Select'

import { TwitterShareButton, FacebookShareButton } from './ShareButton'

const atoms = {
  Text,
  Heading,
  Button,
  IconButton,
  Input,
  RadioGroup,
  RadioButton,
  Select,
  TwitterShareButton,
  FacebookShareButton
}

export default atoms
