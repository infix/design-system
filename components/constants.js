/* global module */
const defaultWeights = [
  400,
  700
]

const defaultBreakpoints = [
  40,
  52,
  64
]

const defaultSpace = [
  0,
  8,
  16,
  32,
  64
]

const defaultFontSizes = [
  12,
  14,
  16,
  20,
  24,
  32,
  48,
  64,
  72
]

module.exports = {
  defaultBreakpoints,
  defaultSpace,
  defaultFontSizes,
  defaultWeights
}
