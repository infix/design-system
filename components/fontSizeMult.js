/* global require */
import {
  style,
  px
} from 'styled-system'

const {
  defaultFontSizes
} = require('./constants')

export const fontSizeMult = (propName, k) => {
  /* eslint-disable no-console */
  console.error('usage of fontSizeMult will be deprecated')
  /* eslint-enable no-console */

  return style({
    prop: 'fontSize',
    cssProperty: propName,
    key: 'fontSizes',
    transformValue: n => px(n * k),
    scale: defaultFontSizes
  })
}

export default fontSizeMult
