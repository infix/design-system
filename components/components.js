import atoms from './atoms'
import molecules from './molecules'
import organisms from './organisms'

import BarChart from './charts/BarChart'
import PieChart from './charts/PieChart'

/* OONI produced atoms */

export const Button = atoms.Button
export const Input = atoms.Input
export const RadioButton = atoms.RadioButton
export const RadioGroup = atoms.RadioGroup
export const TwitterShareButton = atoms.TwitterShareButton
export const FacebookShareButton = atoms.FacebookShareButton
export const Heading = atoms.Heading
export const Text = atoms.Text
export const Select = atoms.Select

/* OONI produced molecules */
export const InputWithIconButton = molecules.InputWithIconButton
export const LogoOONIRun = molecules.LogoOONIRun
export const Card = molecules.Card

/* OONI produced organisms */
export const Hero = organisms.Hero
export const HeroLead = organisms.HeroLead
export const Modal = organisms.Modal

/* Here we just export rebass until we implement all of these */
import {
  Absolute,
  Arrow,
  Avatar,
  BackgroundImage,
  Badge,
  Banner,
  BlockLink,
  Blockquote,
  Border,
  ButtonCircle,
  ButtonOutline,
  ButtonTransparent,
  Carousel,
  CarouselSlide,
  Checkbox,
  Circle,
  Close,
  Code,
  Column,
  Container,
  Divider,
  Donut,
  DotButton,
  Drawer,
  Embed,
  Fixed,
  Group,
  Image,
  Label,
  Lead,
  Link,
  Measure,
  Media,
  Message,
  NavLink,
  Overlay,
  Panel,
  PanelFooter,
  PanelHeader,
  Pre,
  Progress,
  Radio,
  Relative,
  Row,
  Samp,
  ScrollCarousel,
  Slider,
  Small,
  Star,
  Sticky,
  Subhead,
  Switch,
  TabItem,
  Tabs,
  Textarea,
  Toolbar,
  Tooltip,
  Truncate
} from 'rebass'

export { default as BarChart } from './charts/BarChart'
export { default as PieChart } from './charts/PieChart'

const components = {
  Button,
  Heading,
  Text,
  Input,
  Select,
  Card,
  RadioButton,
  RadioGroup,
  InputWithIconButton,
  HeroLead,
  Hero,
  LogoOONIRun,
  TwitterShareButton,
  FacebookShareButton,
  Modal,
  BarChart,
  PieChart,
  Absolute,
  Arrow,
  Avatar,
  BackgroundImage,
  Badge,
  Banner,
  BlockLink,
  Blockquote,
  Border,
  ButtonCircle,
  ButtonOutline,
  ButtonTransparent,
  Carousel,
  CarouselSlide,
  Checkbox,
  Circle,
  Close,
  Code,
  Column,
  Container,
  Divider,
  Donut,
  DotButton,
  Drawer,
  Embed,
  Fixed,
  Group,
  Image,
  Label,
  Lead,
  Link,
  Measure,
  Media,
  Message,
  NavLink,
  Overlay,
  Panel,
  PanelFooter,
  PanelHeader,
  Pre,
  Progress,
  Radio,
  Relative,
  Row,
  Samp,
  ScrollCarousel,
  Slider,
  Small,
  Star,
  Sticky,
  Subhead,
  Switch,
  TabItem,
  Tabs,
  Textarea,
  Toolbar,
  Tooltip,
  Truncate
}

export default components
