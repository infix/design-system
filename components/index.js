import {
  Provider,
  Flex,
  Box
} from 'rebass'

import colors from './theme/colors'
import theme from './theme/rebassTheme'

export { default as colors } from './theme/colors'
export { default as theme } from './theme/rebassTheme'

import components from './components'

// We export these from rebass directly
export {
  Provider,
  Flex,
  Box
} from 'rebass'

export const {
  Button,
  ButtonOutline,
  ButtonCircle,
  ButtonTransparent,
  Link,
  NavLink,
  BlockLink,
  Heading,
  Subhead,
  Text,
  Small,
  Lead,
  Pre,
  Code,
  Samp,
  Blockquote,
  Measure,
  Truncate,
  Label,
  Input,
  Select,
  Textarea,
  Checkbox,
  Radio,
  Slider,
  Image,
  Avatar,
  BackgroundImage,
  Container,
  Divider,
  Border,
  Media,
  Card,
  Banner,
  Panel,
  PanelHeader,
  PanelFooter,
  Progress,
  Message,
  Group,
  Toolbar,
  Badge,
  Circle,
  Tabs,
  TabItem,
  DotButton,
  Close,
  Relative,
  Absolute,
  Fixed,
  Sticky,
  Drawer,
  Overlay,
  Carousel,
  ScrollCarousel,
  CarouselSlide,
  Tooltip,
  Switch,
  Arrow,
  Star,
  Embed,
  Donut,
  Row,
  Column,
  RadioGroup,
  RadioButton,
  HeroLead,
  Hero,
  LogoOONIRun,
  InputWithIconButton,
  TwitterShareButton,
  FacebookShareButton,
  Modal,
  BarChart,
  PieChart
} = components

export const OONIComponents = Object.assign({}, components, { Provider, Box, Flex, colors, theme })

export default OONIComponents
