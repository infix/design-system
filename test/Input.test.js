/* global describe, test, expect, jest */
import React from 'react'
import { renderJSON, renderShallowJSON } from './index'
import { Input } from '../components'

describe('Input', () => {
  test('renders', () => {
    const json = renderJSON(
      <Input error='Some error' />
    )
    expect(json).toMatchSnapshot()
  })

  test('renders with prop type', () => {
    const json = renderShallowJSON(
      <Input type='number' />
    )
    expect(json).toMatchSnapshot()
  })

  test('renders with prop rows', () => {
    const json = renderShallowJSON(
      <Input type='textarea' rows={4} />
    )
    expect(json).toMatchSnapshot()
  })

  test('renders with prop fontSize', () => {
    const json = renderShallowJSON(
      <Input fontSize={4} />
    )
    expect(json).toMatchSnapshot()
  })

  test('renders with onChange handler', () => {
    // const value = 'OONI'
    // const onChange = jest.fn()
    const wrapper = renderShallowJSON(
      <Input onChange={jest.fn()} />
    )
    expect(wrapper).toMatchSnapshot()
    // XXX: Should we simulate an event
    // wrapper.find('input').simulate('change', {
    //   target: { value }
    // })
    // expect(onChange).toBeCalledWith(value)
  })
})
