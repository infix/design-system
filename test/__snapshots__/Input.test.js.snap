// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders 1`] = `
.c2 {
  box-sizing: border-box;
}

.c4 {
  box-sizing: border-box;
  margin-left: auto;
  width: 16px;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c3 {
  font-size: 16px;
  color: #f03e3e;
}

.c0 {
  font-size: 16px;
  font-family: inherit;
  display: 'block';
  vertical-align: 'middle';
  background-color: transparent;
  border: none;
  border-bottom: 2px solid #f03e3e;
  border-radius: 0;
  margin: 0;
  padding: 0;
  outline: none;
  width: 100%;
  height: 32px;
  box-shadow: none;
  box-sizing: content-box;
  -webkit-transition: all .3s;
  transition: all .3s;
}

.c0:hover:not([readonly]) {
  border-bottom: 2px solid #f03e3e;
}

.c0:focus:not([readonly]) {
  border-bottom: 2px solid #f03e3e;
}

<Input
  error="Some error"
  fontSize={1}
  onChange={[Function]}
  rows={3}
  type="text"
>
  <div>
    <Input__StyledInput
      error="Some error"
      fontSize={1}
      onChange={[Function]}
      rows={3}
      type="text"
    >
      <input
        className="c0"
        fontSize={1}
        onChange={[Function]}
        rows={3}
        type="text"
      />
    </Input__StyledInput>
    <ErrorMessage>
      <Flex
        flexWrap="wrap"
      >
        <Box
          className="c1"
          flexWrap="wrap"
        >
          <div
            className="c1 c2"
          >
            <Box>
              <div
                className="c2"
              >
                <Input__StyledErrorMessage
                  fontSize={1}
                >
                  <p
                    className="c3"
                    fontSize={1}
                  >
                    Some error
                  </p>
                </Input__StyledErrorMessage>
              </div>
            </Box>
            <Box
              flex="none"
              ml="auto"
              width={16}
            >
              <div
                className="c4"
                width={16}
              >
                <Input__StyledErrorMessage
                  fontSize={1}
                >
                  <p
                    className="c3"
                    fontSize={1}
                  >
                    <MdWarning>
                      <IconBase
                        viewBox="0 0 40 40"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          preserveAspectRatio="xMidYMid meet"
                          style={
                            Object {
                              "color": undefined,
                              "verticalAlign": "middle",
                            }
                          }
                          viewBox="0 0 40 40"
                          width="1em"
                        >
                          <g>
                            <path
                              d="m21.6 23.4v-6.8h-3.2v6.8h3.2z m0 6.6v-3.4h-3.2v3.4h3.2z m-20 5l18.4-31.6 18.4 31.6h-36.8z"
                            />
                          </g>
                        </svg>
                      </IconBase>
                    </MdWarning>
                  </p>
                </Input__StyledErrorMessage>
              </div>
            </Box>
          </div>
        </Box>
      </Flex>
    </ErrorMessage>
  </div>
</Input>
`;

exports[`Input renders with onChange handler 1`] = `
<div>
  <Input__StyledInput
    error={null}
    fontSize={1}
    onChange={[MockFunction]}
    rows={3}
    type="text"
  />
</div>
`;

exports[`Input renders with prop fontSize 1`] = `
<div>
  <Input__StyledInput
    error={null}
    fontSize={4}
    onChange={[Function]}
    rows={3}
    type="text"
  />
</div>
`;

exports[`Input renders with prop rows 1`] = `
<div>
  <Input__StyledTextarea
    error={null}
    fontSize={1}
    onChange={[Function]}
    rows={4}
    type="textarea"
  />
</div>
`;

exports[`Input renders with prop type 1`] = `
<div>
  <Input__StyledInput
    error={null}
    fontSize={1}
    onChange={[Function]}
    rows={3}
    type="number"
  />
</div>
`;
